grape.py

```diff
@@ -48,16 +48,32 @@ def grape(H0, Hk, u_0, rho_0, C, T, alpha, target="trace_real", epsilon=1e-6, ma
             break
 
         # last phi
-        phi = np.trace(np.dot( C.T.conjugate(), rhoj[-1] ))
+        phi = np.trace(C.T.conj() @ rhoj[-1])
+        if target == "trace_real":
+            pass
+        elif target == "trace_both":
+            phi = np.real(phi)

```

Comment:
The way you calculate phi with non-hermitian operator is not correct. Think about how to decompose `C` in to a hermitian `C_x` and a anti-hermitain `C_y`